### PR TITLE
Fix #5820

### LINKED ISSUE #5820: [BUG]: pass selection dropdown no longer auto-scrolls to current pass in GCC Tree/RTL viewer
### Describe the bug

Issue #4444 has reappeared once again.

In GCC Tree/RTL Viewer, pass selection dropdown used to autoscroll to the currently selected entry, so selecting previous/next passes was easy. Now it's no longer the case. This seems to be a recent regression.

### Steps to reproduce

1.    Visit godbolt.org
2.    Select gcc in compiler pane.
3.    Press Add New > GCC Tree/RTL.
4.    Select one of the later passes.
5.    Activate the selector again.
6.    Observe that the selector is on the first pass, and the previously chosen pass is not in view.

### Expected behavior

At step 6, the dropdown should show the selected entry.

### Reproduction link

https://godbolt.org/z/z7Kj4qecE

### Screenshots

Not applicable

### Operating System

_No response_

### Browser version

Firefox 118

### PR DESCRIPTION
**Pull request by OfekShilon**
The same dropdown glitch described in the bug happens for the compiler selection dropdown. This addresses both.

### CODE REVIEW

**Comment by OfekShilon** [comment]
The call to `selecize.clearOptions()` had a lot of indirect negative side-effects inside tom-select. In a nutshell, all options had to be re-created but with different `#order` attributes, but the CE-stored selection still referred to the old option. In such a state [tom-select reset the selected option to 0](https://github.com/orchidjs/tom-select/blob/69180fa9e79060060f1824fbde85537579d0005d/src/tom-select.ts#L1588). 

Anyway removing this call causes the dropdown to behave as expected.

File: static/panes/gccdump-view.ts [hunk 1]
```diff
@@ -235,6 +235,24 @@ export class GccDump extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Gcc
...
+            if (!this.selectedPass) return;
+
+            // Make the trip back from the stored selectedPass to the TomSelect option
+            const activeOption = Object.entries(this.selectize.options).find(
+                op => op[1].filename_suffix === this.selectedPass,
+            );
+            const selectedPassId = activeOption![0];
+            const option = this.selectize.getOption(selectedPassId);
+
+            this.selectize.setActiveOption(option);
```
[the remark this answers is not in the document]
Comment (jeremy-rifkin) [reply static/panes/gccdump-view.ts:254]: I see. How do we handle this for other tomselect menus?

[the remark this answers is not in the document]
Comment (jeremy-rifkin) [reply static/panes/gccdump-view.ts:254]: Looking at the live site it looks like some of our other tomselect menus have issues too. As far as expected behavior goes, either scrolling to the selected item or going to the previous scroll position seems reasonable to me.

File: static/widgets/compiler-picker.ts [hunk 2]
```diff
@@ -172,6 +172,11 @@ export class CompilerPicker {
                 this.popup.show();
             });
             this.popupTooltip.appendTo(this.toolbarPopoutButton);
+
+            // TomSelect's default behavior is to restore the last scroll position. We wish to scroll to the active
+            // selection.
+            const selection = this.tomSelect!.getOption(this.lastCompilerId);
+            this.tomSelect!.setActiveOption(selection);
```
[the remark this answers is not in the document]
Comment (mattgodbolt) [reply static/widgets/compiler-picker.ts:179]: Or is this needed at all? How might the `tomSelect` be undefined here? The whole section is inside a `this.tomSelect.on()` block, so we know tomSelect is defined. And if it's been `destroy()`ed and this code is still able to run, then don't we have other problems? I don't mind defensivenes in general but sometimes it becomed "FUD" that can leave you wondering what things are reasonable and what not :)

File: static/panes/gccdump-view.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattgodbolt) [reply static/panes/gccdump-view.ts:254]: I'm not against our own fork of TomSelect, but there's a lot of code there and a lot of other fixes I'd like to pick up (eventually...) For example, TomSelect is the only reason we haven't upgraded to TypeScript 5. I tried looking at the work involved in fixing it and decided I'd wait for @oyejorge to take a look as we don't have time to fix. But if it's something you think is worth branching I'm happy to do so (so long as we track upstream!)

**Review by mattgodbolt — APPROVED** [review]
I'm happy with the ideas here, and as you know, I'm happier to get things into the wild and tested more than letting them sit as PRs for toooo long :)

**Comment by mattgodbolt** [comment]
Have drafted this so we don't accidentally mere but @OfekShilon just undraft when you're ready! Thank you so much :)

**Comment by OfekShilon** [comment]
The TomSelect behavior fixes discussed above are now applied to cfg-view, device_view, editor, gccdump-view, opt-pipeline, tree and compiler-picker. NOT applied yet to diff.ts (that got hairy and is probably low priority anyway). 
As far as I can see this is ready to merge, and reviews are always welcome.

**Comment by mattgodbolt** [comment]
Thanks Ofek!

**Comment by mattgodbolt** [comment]
This is now live

**Pull request merged**